message PlayerMove {
  required float mvectx = 2;
  required float mvecty = 3;
}

message PlayerChange {
  required int32 time = 1;
  required float angle = 2;
  optional PlayerMove move = 3;
  optional int32 firedNumber = 4;
  repeated float firedAngle = 5 [packed=true];
}

message PlayerUpdate {
  required int32 player = 1;
  required float angle = 2;
  required float posx = 3;
  required float posy = 4;
  required float velx = 5;
  required float vely = 6;
  optional int32 firedNumber = 7;
  repeated float firedAngle = 8 [packed=true];
}

message Update {
  required int64 time = 1;
  repeated PlayerUpdate players = 2;
}

option optimize_for=LITE_RUNTIME;